### PR TITLE
Emulates implicit submission on each form page.

### PR DESCRIPTION
**Pull request by gangaramani-nikhil**
Implicit submission mapped enter to next or finish button

### CODE REVIEW

**Comment by gangaramani-nikhil** [comment]
Hi @trevorshannon ,
I have created this pull request for issue #276

**Review by trevorshannon — CHANGES_REQUESTED** [review]

File: src/site/_includes/js/eligibility.js
[the remark this answers is not in the document]
Comment (trevorshannon) [reply src/site/_includes/js/eligibility.js:1]: Oh sorry, typo above: `<input>` or `<select>`

**Comment by gangaramani-nikhil** [comment]
Hi @trevorshannon,
On the first page do you want me to map enter to next button?
There is no input or select element on first page
I am referring to the following page
image

File: src/site/_includes/js/eligibility.js [hunk 1]
```diff
@@ -664,7 +664,12 @@ function addListeners() {
   document.getElementById("next-button").addEventListener("click", toNextPage);
   document.getElementById("back-button").addEventListener("click", toPrevPage);
   document.getElementById("submit-button").addEventListener("click", submitForm);
-
+  document.addEventListener("keypress", function(event) {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      document.getElementById("next-button").click();
```
[the remark this answers is not in the document]
Comment (gangaramani-nikhil) [reply src/site/_includes/js/eligibility.js]: sure updated in latest commit

File: src/site/_includes/js/eligibility.js
[the remark this answers is not in the document]
Comment (gangaramani-nikhil) [reply src/site/_includes/js/eligibility.js:1]: solved this issue Thanks

**Comment by trevorshannon** [comment]
In this case, as there is no input field to have focus, it would not advance when the user presses Enter.

**Comment by gangaramani-nikhil** [comment]
yep that is how it is implemented now

File: src/site/_includes/js/eligibility.js [hunk 2]
```diff
@@ -664,7 +664,17 @@ function addListeners() {
   document.getElementById("next-button").addEventListener("click", toNextPage);
   document.getElementById("back-button").addEventListener("click", toPrevPage);
   document.getElementById("submit-button").addEventListener("click", submitForm);
-
+  document.addEventListener("keypress", function(event) {
+    if ( event.key === "Enter" && ( document.activeElement.tagName === "INPUT" || document.activeElement.tagName === "SELECT" ) ){
```
Comment (trevorshannon) [inline src/site/_includes/js/eligibility.js:668]: We generally try to follow the Google JS style guide, which recommends omitting spaces just inside the parentheses.

```
if (thisIsGood) {

if ( notRecommended ) {
```

https://google.github.io/styleguide/jsguide.html#formatting-horizontal-whitespace

File: src/site/_includes/js/eligibility.js [hunk 3]
```diff
@@ -664,7 +664,17 @@ function addListeners() {
...
+      event.preventDefault();
+      if(document.getElementById("next-button").className !== "btn btn_primary hidden") {
+        document.getElementById("next-button").click();
+      }
+      else{
+        document.getElementById("submit-button").click();
+      }
+    }
+  })
```
Comment (trevorshannon) [inline src/site/_includes/js/eligibility.js:677]: Add a semicolon here, please.

**Review by trevorshannon — CHANGES_REQUESTED** [review]

**Comment by gangaramani-nikhil** [comment]
Hi @trevorshannon ,
I have made the changes.

**Review by trevorshannon — APPROVED** [review]

**Pull request merged**